pulser-core/pulser/sequence/_schedule.py

```diff
@@ -392,6 +410,14 @@ def add_pulse(
         ti = t0 + delay_duration
         tf = ti + pulse.duration
         self._check_duration(tf)
+        # dataclasses.replace() does not work on Pulse (because init=False)
+        if phase_drift_params is not None:
+            pulse = Pulse(
+                amplitude=pulse.amplitude,
+                detuning=pulse.detuning,
+                phase=corrected_phase(ti),
```

Comment:
If I get it correctly, this apply an additional phase that compensate the phase drift acquired by the non-zero detuning delays between EOM pulses.
That way there is no need for the user to track it.

Reply:
Exactly. Furthermore, since the compensation is only computed at runtime, it will also work for parametrized sequences.

Reply:
Nice! =)